presto-spi/src/main/java/com/facebook/presto/spi/security/GrantInfo.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.security;
+
+import com.facebook.presto.spi.SchemaTableName;
+
+import java.util.Objects;
+import java.util.Set;
+
+public class GrantInfo
+{
+    private final Set<PrivilegeInfo> privilegeInfo;
+    private final Identity identity;
+    private final SchemaTableName schemaTableName;
+
+    public GrantInfo(Set<PrivilegeInfo> privilegeInfo, Identity identity, SchemaTableName schemaTableName)
```

Comment:
hmm interesting. Theoretically you should be able to ask for all of the grants a user has across all of the catalogs -- SHOW GRANT ON ALL seems like it should return that.